packages/dht/src/connection/WebRTC/WebRtcConnector.ts

```diff
@@ -0,0 +1,274 @@
+    constructor(private config: WebRtcConnectorConfig) {
+        super()
+        this.rpcTransport = config.rpcTransport
+
+        this.rpcCommunicator = new RoutingRpcCommunicator(this.WEBRTC_CONNECTOR_APP_ID, this.rpcTransport, {
+            rpcRequestTimeout: 10000
+        })
+
+        this.getManagerConnection = config.fnGetConnection
+        this.addManagerConnection = config.fnAddConnection
+
+        this.rtcOffer = this.rtcOffer.bind(this)
+        this.rtcAnswer = this.rtcAnswer.bind(this)
+        this.iceCandidate = this.iceCandidate.bind(this)
+        this.requestConnection = this.requestConnection.bind(this)
+
+        this.rpcCommunicator.registerRpcNotification(RtcOffer, 'rtcOffer', this.rtcOffer)
+        this.rpcCommunicator.registerRpcNotification(RtcAnswer, 'rtcAnswer', this.rtcAnswer)
+        this.rpcCommunicator.registerRpcNotification(IceCandidate, 'iceCandidate', this.iceCandidate)
+        this.rpcCommunicator.registerRpcNotification(WebRtcConnectionRequest, 'requestConnection', this.requestConnection)
+    }
+
+    connect(targetPeerDescriptor: PeerDescriptor): IConnection {
+        if (!PeerID.fromValue(this.ownPeerDescriptor!.peerId).equals(PeerID.fromValue(targetPeerDescriptor.peerId))) {
+            logger.trace(`Opening WebRTC connection to ${targetPeerDescriptor.peerId.toString()}`)
+            const existingConnection = this.getWebRtcConnection(targetPeerDescriptor)
+            if (existingConnection) {
+                return existingConnection as unknown as IConnection
+            }
+            setImmediate(() => {
+                const newConnection = this.createConnection(targetPeerDescriptor)
+                const added = this.addManagerConnection(targetPeerDescriptor, newConnection)
+                if (!added) {
+                    newConnection.close()
+                }
+            })
+            return new DeferredConnection(targetPeerDescriptor)
+        }
+        throw new Err.CannotConnectToSelf('Cannot open WebRTC Connection to self')
+    }
+
+    setOwnPeerDescriptor(peerDescriptor: PeerDescriptor): void {
+        this.ownPeerDescriptor = peerDescriptor
+    }
+
+    getWebRtcConnection(peerDescriptor: PeerDescriptor): NodeWebRtcConnection | null {
+        const connection = this.getManagerConnection(peerDescriptor)
+        if (connection && connection.connectionType === ConnectionType.WEBRTC) {
+            return connection as unknown as NodeWebRtcConnection
+        }
+        return null
+    }
+
+    private createConnection(targetPeerDescriptor: PeerDescriptor, sendRequest = true): NodeWebRtcConnection {
+        const connection = new NodeWebRtcConnection(targetPeerDescriptor)
+        this.bindListenersAndStartConnection(targetPeerDescriptor, connection, sendRequest)
+        return connection
+    }
+
+    private onRtcOffer(
+        remotePeerDescriptor: PeerDescriptor,
+        targetPeerDescriptor: PeerDescriptor,
```

Comment:
```suggestion
        remotePeer: PeerDescriptor,
        targetPeer: PeerDescriptor,
```

We could consider shortening some variable names which contain "type information" as a suffix. (This snippet just an example.)